backend/homework/1/app/src/main/java/Main.java

```diff
@@ -0,0 +1,120 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.*;
+
+import java.util.*;
+
+public class Main {
+
+//    private static final Logger logger
+//            = LoggerFactory.getLogger(logger.class);
+
+    public static void main(String args[]) {
+
+        StudentRepo stu = new StudentRepo(5);
+        Scanner scanner = new Scanner(System.in);
+
+
+
+        System.out.println("sd");
+
+        int choice = 1;
+        do {
+            System.out.println("Choose an operation:");
+            System.out.println("1. Add Student");
+            System.out.println("2. Update Student Information");
+            System.out.println("3. Retrieve Student by ID");
+            System.out.println("4. Retrieve Student by Name");
+            System.out.println("0. Exit");
+
+            System.out.print("Enter your choice: ");
+            choice = scanner.nextInt();
+
+            switch (choice) {
+                case 1:
+
+                    System.out.print("Enter student ID: ");
+                    int id = scanner.nextInt();
+
+                    System.out.print("Enter student Age: ");
+                    int age = scanner.nextInt();
+
+                    System.out.print("Enter student Name: ");
+                    String name = scanner.next();
+
+                    System.out.print("Enter student Grade: ");
+                    String grade = scanner.next();
+                    Student std = new Student(id, age, name, grade);
+
+                    stu.addStudent(std);
+                    logger.msgreceive().info("Student added successfully");
+                    break;
+                case 2:
+                    System.out.println("Enter the id for updation");
+                    int idd = scanner.nextInt();
+                    System.out.print("Enter student ID: ");
+                    int id2 = scanner.nextInt();
+
+                    System.out.print("Enter student Age: ");
+                    int age2 = scanner.nextInt();
+
+                    System.out.print("Enter student Name: ");
+                    String name2 = scanner.next();
+
+                    System.out.print("Enter student Grade: ");
+                    String grade2 = scanner.next();
+
+
+                    Student std1 = new Student(id2, age2, name2, grade2);
+
+                    stu.updateStudent(std1, idd);
```

Comment:
How come idd is related to idx of student? I didn't get how you are managing this case.